File: app/src/main/java/com/mouredev/weeklychallenge2022/Challenge50_test.rb
# frozen_string_literal: true

gem 'minitest', '~> 5.4'
require 'minitest/autorun'
require 'minitest/pride'
require_relative './Challenge50'

# class for testing
class KaracaTest < Minitest::Test
  def test_encryption
    # skip
    input = { input: 'apple', func: 'encryption' }
    expected = '1lpp0aca'
    assert_equal expected, Karaca.new(input).start
  end

  def test_decryption
    # skip
    input = { input: '0n0n0baca', func: 'decryption' }
    expected = 'banana'
    assert_equal expected, Karaca.new(input).start
  end

  def test_encryption_sentence
    # skip
    input = { input: 'this is a long sentence', func: 'encryption' }
    expected = 's2htaca s2aca 0aca gn3laca 1cn1tn1saca'
    assert_equal expected, Karaca.new(input).start
  end

  def test_decryption_sentence
    # skip
    input = { input: 't0htaca s0waca 0aca gn3laca 1cn1tn1saca', func: 'decryption' }
    expected = 'that was a long sentence'
    assert_equal expected, Karaca.new(input).start
  end

  def test_decryption_not_encrypted
    # skip
    input = { input: 'this is a long sentence', func: 'decryption' }
    expected = 'not encrypted input'
    assert_equal expected, Karaca.new(input).start
  end

  def test_decryption_not_encrypted_by_one
    # skip
    input = { input: 't0htaca s0waca 0 gn3laca 1cn1tn1saca', func: 'decryption' }
    expected = 'not encrypted input'
    assert_equal expected, Karaca.new(input).start
  end
end

File: app/src/main/java/com/mouredev/weeklychallenge2022/Challenge50_test.js
const karaca = require("./Challenge50");

const tests = {
  input: [
    { input: "apple", func: "encryption" },
    { input: "0n0n0baca", func: "decryption" },
    { input: "this is a long sentence", func: "encryption" },
    { input: "t0htaca s0waca 0aca gn3laca 1cn1tn1saca", func: "decryption" },
    { input: "this is a long sentence", func: "decryption" },
    { input: "t0htaca s0waca 0 gn3laca 1cn1tn1saca", func: "decryption" },
  ],
  output: [
    "1lpp0aca",
    "banana",
    "s2htaca s2aca 0aca gn3laca 1cn1tn1saca",
    "that was a long sentence",
    "not encrypted input",
    "not encrypted input",
  ],
};

let errors = 0;
tests.input.forEach((test, index) => {
  const testFunction = karaca.func(test);
  const expected = tests.output[index];
  if (testFunction != expected) {
    errors += 1;
    console.log("\noriginal:", test);
    console.log(testFunction);
    console.log("expected:", expected);
  }
});

console.log(`Tests${errors != 0 ? " not " : " "}passed, ${errors} errors`);
